storage/range_command.go

```diff
@@ -864,6 +864,17 @@ func (r *Range) AdminSplit(args *proto.AdminSplitRequest, reply *proto.AdminSpli
 		return txn.Flush()
 	}); err != nil {
 		reply.SetGoError(util.Errorf("split at key %s failed: %s", splitKey, err))
+		return
+	}
+
+	if err := r.rm.ProcessRangeDescriptorUpdate(r.Desc().RaftID); err != nil {
```

Comment:
I see. Thanks, Ben. Yes, calling ProcessRangeDescriptorUpdate in Range.SetDesc worked. I made the following changes:

- Make SetDesc() return an error.
- Change NewRange() to directly set r.desc to avoid calling ProcessRangeDescriptorUpdate() from there.
- Remove the comment on metalock from SetDesc(). I think this is not true. We call SetDesc() without Store.mu.Lock.